src/components/MainNavigation.tsx

```diff
@@ -0,0 +1,160 @@
+      dispatch(setProfile(data));
+      return data;
+    },
+    enabled: isLoggedIn && userId !== null,
+  });
+
+  const { mutate } = useMutation({
+    mutationKey: ['logout'],
+    mutationFn: async () => {
+      await httpClient.post('/auth/signout');
+    },
+    onSuccess: () => {
+      dispatch(logout());
+      queryClient.removeQueries();
+      navigate('/');
+    },
+  });
+
+  const handleClose = (action: string) => {
+    switch (action) {
+      case 'account':
+        navigate('/account');
+        break;
+      case 'logout':
+        mutate();
+        break;
+      case 'signIn':
+        navigate('auth/signin');
+        break;
+      case 'signUp':
+        navigate('auth/signup');
+        break;
+      default:
+        break;
+    }
+
+    setAnchorEl(null);
+  };
+
+  return (
+    <Container maxWidth={maxWidth} disableGutters sx={{ padding: '1rem' }}>
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+        }}
+      >
+        <Box
+          sx={{
+            'display': 'flex',
+            'alignItems': 'center',
+            '& img': {
+              width: '2.5rem',
+              height: '2.5rem',
+            },
+          }}
+        >
+          <Link to="/" style={{ textDecoration: 'none' }}>
+            <Box sx={{ display: 'flex', alignItems: 'center' }}>
+              <img src={logo} alt="logo" />
+              <Typography
+                variant="h5"
+                sx={{
+                  marginLeft: '1rem',
+                  color: 'secondary.main',
+                  fontWeight: 'bold',
+                  marginX: '0.5rem',
+                }}
+              >
+                airbnb
+              </Typography>
+            </Box>
+          </Link>
+        </Box>
+        <Button
+          id="basic-button"
+          aria-controls={open ? 'basic-menu' : undefined}
+          aria-haspopup="true"
+          aria-expanded={open ? 'true' : undefined}
+          onClick={handleClick}
+          sx={{
+            'display': 'flex',
+            'alignItems': 'center',
+            'justifyContent': 'space-between',
+            'textTransform': 'none',
+            'borderRadius': '2rem',
+            'borderColor': 'secondary2.main',
+            'color': 'secondary2.main',
+            'borderWidth': '1px',
+            'borderStyle': 'solid',
+            '& svg': {
+              marginLeft: '0.5rem',
+            },
+          }}
+        >
+          <MenuIcon />
+          <AccountCircleIcon fontSize="large" />
+        </Button>
+        <Menu
+          id="basic-menu"
+          anchorEl={anchorEl}
+          open={open}
+          onClose={handleClose}
+          MenuListProps={{
+            'aria-labelledby': 'basic-button',
+          }}
+        >
+          {isLoggedIn && !isError
+            ? [
+                <MenuItem key="account" onClick={() => handleClose('account')}>
```

Comment:
don't use lambdas in handlers. This approach creates new function for every render. moreover, it's better prachice to use HashMap instead of "switch case"
onClick={ handleClose.account}
const handleClose = useMemo({
   account: () => navigate('/account'),
   logout: () => ...
}, []
)

but in your case you may donn't need to join clicks on 'account' and 'signin/signup' into one structure. Just need several separate memoized functions

Reply:
functions should be wrapped with useCallback to memoize it